### PR TITLE
Bhbc 865 2

### PR DESCRIPTION
**Pull request by NickPhura**
# Overview

## Links to jira tickets

https://quartech.atlassian.net/browse/BHBC-865

## This PR contains the following changes

Mainly the backend changes for the latest permit updates
- Can submit a project when there are only permits with no sampling
   - new endpoint `/permit-no-sampling` 
   - Creates rows in the `no_sample_permit` table

On the create project page:
- Initially should load and show all steps
- If no permits are specified, user can continue to next steps
- if permits are specified, but none of them have sampling conducted = true, then cut the workflow short and let them save just the permits/coordinator info

## This PR contains the following types of changes

- [x] New feature (change which adds functionality)
- [ ] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [ ] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [x] I have performed a self-review of my own code

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [x] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [x] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [x] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [x] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

**Comment by NickPhura** [comment]
I need to update the postman requests, but will tackle that tomorrow (Friday)

File: app/src/features/projects/CreateProjectPage.test.tsx [hunk 1]
```diff
@@ -45,52 +45,75 @@ describe('CreateProjectPage', () => {
 
       expect(getByText('Permits')).toBeVisible();
 
+      expect(getByText('General Information')).toBeVisible();
```
Comment (sdevalapurkar) [inline app/src/features/projects/CreateProjectPage.test.tsx:48]: duplicate assertion :)

File: app/src/features/projects/CreateProjectPage.test.tsx [hunk 2]
```diff
@@ -45,52 +45,75 @@ describe('CreateProjectPage', () => {
...
 
-    // wait for project coordinator form to load
-    expect(await findByText('First Name')).toBeVisible();
+    // wait for initial page to load
+    await waitFor(() => {
+      expect(getByText('Project Coordinator')).toBeVisible();
+
+      expect(getByText('Permits')).toBeVisible();
+
+      expect(getByText('General Information')).toBeVisible();
```
Comment (sdevalapurkar) [inline app/src/features/projects/CreateProjectPage.test.tsx:78]: same

**Comment by NickPhura** [comment]
Closing to reopen with adjusted branch name

File: app/src/features/projects/CreateProjectPage.test.tsx [hunk 3]
```diff
@@ -45,52 +45,75 @@ describe('CreateProjectPage', () => {
...
-    fireEvent.change(getByTestId('sampling_conducted'), { target: { value: 'true' } });
+    fireEvent.change(getByTestId('sampling_conducted'), { target: { value: 'false' } });
 
     // wait for forms to load
     await waitFor(() => {
-      expect(getByText('General Information')).toBeVisible();
-
       expect(getByText('Project Coordinator')).toBeVisible();
 
       expect(getByText('Permits')).toBeVisible();
```
Comment (sdevalapurkar) [inline app/src/features/projects/CreateProjectPage.test.tsx:115]: the snapshot match is good, but can we add an assertion here to say that

```
expect(queryByText('General Information')).toBeNull();
```

**Pull request closed without merging**